### ISSUE TITLE
Demo is broken with IE : Inform users about problems of compatibility with their browser 

### ISSUE THREAD

**Issue by gmaillet** [IGNF/itowns-legacy#27]
The demo doesn't work with IE: it would be nice to inform the user and to suggest to change his browser.

**Comment by tbroyer** [comment]
FWIW, the website at http://itowns.github.io/ will inform the user, but indeed not the demos themselves.

**Comment by mbredif** [comment]
I have just put a more informative title. could you please provide some info on the incompatibility (error message, console log, https://get.webgl.org/ status...)

**Comment by vcoindet** [comment]
- IE 11 doesn't seem to support "Promise()"
- Console.log : SCRIPT5009: "Promise" is undefined (file itowns.js, l.18)
- https://get.webgl.org/ status : IE supports WebGL

For now I propose an alert box to warn users that itowns is not supported by IE and invite them to use chrome or firefox on windows.

**Comment by nosy-b** [comment]
How can we make a automatic conformance test on the builds/release? @tbroyer

**Comment by gcebelieu** [comment]
Maybe this can fill the gap between ie11 and promise : https://github.com/stefanpenner/es6-promise

**Comment by tbroyer** [comment]
@nosy-b Depends what you want to test :-)

ESLint 2.x would choke on the `new Promise()` we have here, but it would choke on so many other things too that it would require time to be configured; or we could start with reformatting the code using, e.g. JS-Beautify, and dumbing down a few ESLint rules to mere warnings rather than errors.
…or we start "from scratch" and only enable the few rules we want (`no-undef` here)

Another solution is to run automated tests in browsers (through SauceLabs' OpenSauce), but that means such tests must exists (could possibly be as easy as running the sample and looking for errors, still this would have to be automated somehow)

@gcebelieu this file imports `when`, which implements a form of promises AFAICT (although different from "native" promises).

**Comment by vpicavet** [comment]
I would say we do not support IE at all for the moment in itowns 1, just alert the user
As for a test framework, probably better to concentrate on itowns2 as well. Let's open issues there.

**Comment by vcoindet** [comment]
I agree. I've already made an alert message to warn users that iTowns is not supported by IE when they are trying to open the project with IE.
Just have to push it.

**Comment by tbroyer** [comment]
If supporting IE11 is only a matter of fixing that `Promise`, I think we should do it (fun fact: in my Windows 7 VM in Virtual Box, IE11 has WebGL but neither Firefox nor Chrome has it). Older IE versions don't have WebGL anyway (and moreover are no longer supported by Microsoft).

Re. test automation et al. in V2, see https://github.com/iTowns/itowns2/pull/45. I'll setup Travis-CI as soon as it's merged in (we could also use AppVeyor for a basic Windows build BTW).

**Comment by gcebelieu** [comment]
IE11 support may be optional for itowns v1, but it isn't for itowns v2 if we want to use it for geoportal web site. We will have to fix it anyway.
